tests/UnitTests/Controllers/ProductControllerTests.cs

```diff
@@ -0,0 +1,197 @@
+        }
+
+        [Test]
+        public async Task AddProduct_Returns_CorrectModels()
+        {
+            // Arrange
+            var expectedProductOutbound = new ProductOutbound();
+            _productServiceMock.Setup(x => x.AddItem(It.IsAny<ProductInbound>(),
+                It.IsAny<CancellationToken>())).ReturnsAsync(expectedProductOutbound);
+
+            // Act
+            var result = await _productController.AddProduct(It.IsAny<ProductInbound>());
+
+            // Assert
+            var createdResult = result as CreatedAtActionResult;
+            Assert.IsAssignableFrom<CreatedAtActionResult>(result);
+            Assert.IsNotNull(createdResult);
+
+            var actualResult = createdResult.Value as ProductOutbound;
+            Assert.IsInstanceOf<ProductOutbound>(actualResult);
+            Assert.AreEqual(expectedProductOutbound, actualResult);
+            _productServiceMock.Verify(x => x.AddItem(It.IsAny<ProductInbound>(), It.IsAny<CancellationToken>()), Times.Once);
+        }
+
+        [Test]
+        public async Task AddProduct_Returns_CorrectProductValues()
+        {
+            // Arrange
+            var productInbound = new ProductInbound
+            {
+                Name = $"The Three Musketeers {DateTime.UtcNow.Ticks}",
+                Description = $"You have likely heard of The Three Musketeers! {DateTime.UtcNow.Ticks}",
+                Author = $"Alexandre Dumas {DateTime.UtcNow.Ticks}",
+                Price = 12.56f,
+                ImageUrl = $"ftp://book.shop/{DateTime.UtcNow.Ticks}/image.jpg"
+            };
+
+            var expectedOutbound = new ProductOutbound
+            {
+                Id = Guid.NewGuid(),
+                Name = productInbound.Name,
+                Description = productInbound.Description,
+                Author = productInbound.Author,
+                Price = productInbound.Price,
+                ImageUrl = productInbound.ImageUrl,
+                BookingId = null
+            };
+
+            _productServiceMock.Setup(x => x.AddItem(productInbound, It.IsAny<CancellationToken>()))
+                               .ReturnsAsync(expectedOutbound);
+
+            // Act
+            var result = await _productController.AddProduct(productInbound);
+            var actualResult = ((CreatedAtActionResult)result).Value as ProductOutbound;
+
+            // Assert
+            Assert.IsNotNull(actualResult);
+            Assert.AreEqual(expectedOutbound.Id, actualResult.Id);
+            Assert.AreEqual(expectedOutbound.Name, actualResult.Name);
+            Assert.AreEqual(expectedOutbound.Description, actualResult.Description);
+            Assert.AreEqual(expectedOutbound.Author, actualResult.Author);
+            Assert.AreEqual(expectedOutbound.Price, actualResult.Price);
+            Assert.AreEqual(expectedOutbound.ImageUrl, actualResult.ImageUrl);
+            Assert.AreEqual(expectedOutbound.BookingId, actualResult.BookingId);
```

Comment:
```suggestion
            actualResult.Should().BeEquivalentTo(expectedOutbound);
```

Not for this PR. But this would look much better if FluentAssertions was used.
I think this is not achievable with regular Assert, without using specific Equals methods.